### ISSUE TITLE
Add ability not to have public props field

### ISSUE THREAD

**Issue by Quijx** [felangel/equatable#50]
**Is your feature request related to a problem? Please describe.**
Say I want to create an equatable class that is part of a public API. In that case I would not want to have the public field `props` exposed because that may confuse the user.

**Describe the solution you'd like**
One solution for immutable classes would be to have an abstract class `ImmutableEquatable` that has a private `_props` field and has the list passed in the constructor. For mutable classes this might work by passing a `List<Object> Function()` to the super constructor.

**Describe alternatives you've considered**
The approach above does not not work when the class needs to extend another class, since mixins cannot declare constructors. Because of this it might be good to expose the functions in `equatable_utils.dart` to be able to easily override `hashCode` and `==` yourself without having a public `props` attribute. This would also allow to easily implement an abstract class like `ImmutableEquatable` yourself.

**Comment by hacker1024** [comment]
Would the [`protected`](https://pub.dev/documentation/meta/latest/meta/protected-constant.html) annotation suffice?

**Comment by LeonardoRosaa** [comment]
This could be too expensive to do and also I think there are Dart limitations. However, you can use the `@protected` annotation:

```dart
class MyClass extends Equatable {

  const MyClass();

  @protected
  @override
  List<Object?> get props => [];
}
```

**Comment by felangel** [comment]
Closing this as it will be resolved by #175.
